dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/metadata/version/MetadataVersionDelegateTest.java

```diff
@@ -417,9 +438,26 @@ public void testShouldNotDownloadMetadataVersion()
         AvailabilityStatus availabilityStatus = new AvailabilityStatus( true, "test_message", null );
 
         when( synchronizationManager.isRemoteServerAvailable() ).thenReturn( availabilityStatus );
-        String actualMetadataVersionSnapshot = target.downloadMetadataVersionSnapshot( metadataVersion );
 
-        assertNull( actualMetadataVersionSnapshot );
+        // TODO not sure this tests what it says. If I understand the intent its
```

Comment:
I need some help deciding what to do with this test.

It says `testShouldNotDownloadMetadataVersion` but the way it was setup it was calling the URL, so trying to download the metadata version (if I understand the code/intent correctly).

Should this be about the remoteServer not being available?  `if ( !(remoteServerAvailable.isAvailable()) )`
If so, I would change the above AvailabilityStatus to false

I just need someone else to double check my assumptions/analysis

Reply:
or it might be about testing `isValidDhisHttpResponse` ? 🤷🏻

Reply:
feel free to invite another reviewer if someone else might now this better @enricocolasante @jbee thank you! 😄

Reply:
I ran the test using coverage, used my best guess at what it was trying to do and adapted it.

So I think what it wanted was to make sure we do not make the HTTP call when the remote server is unavailable. This is what I adapted the test to do.

I also added another test to cover an uncovered case.